### ISSUE TITLE
Better error handling/dependency check

### ISSUE THREAD

**Issue by rmoriz** [ib-controller/ib-controller#79]
So many issues are related to missing or changed filenames of .jar files and are repeated over and over. How about adding a a check with a sane error message instead of

```
17:39:14:277 IBController: IBControllerServer listening on address: 95bce3546271/172.17.0.6 port: 7462
17:39:14:344 IBController: TWS settings directory is /opt/IBJts/jars
Exception in thread "main" java.lang.NoClassDefFoundError: jxmlable/DontObfuscate
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at ibcontroller.IBController.startTws(Unknown Source)
	at ibcontroller.IBController.startTwsOrGateway(Unknown Source)
	at ibcontroller.IBController.load(Unknown Source)
	at ibcontroller.IBController.main(Unknown Source)
Caused by: java.lang.ClassNotFoundException: jxmlable.DontObfuscate
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 16 more
```

so we all save a lot of lifetime.

**Comment by rlktradewright** [comment]
I've been sitting here for ages wondering how best to answer this, and the problem is that I'm really not sure what sort of thing would satisfy you.

Errors like the one you show only occur because users don't follow the instructions for setting up IBController. I don't know why this is, because it is incredibly simple nowadays, but somehow people seem to want to make unjustified assumptions about things, or they want it to be much more complicated than it is, or they want to run before they can walk by setting up some complex configuration when they haven't even tried to do a simple setup first just to make sure it works and to try and understand how it fits together.

To successfully set up IBController 3.0.1, assuming you've downloaded and installed the offline version (I repeat the OFFLINE version! Again, please, the OFFLINE version, not the self-updating version - I keep telling people this over and over again and still people don't listen! INSTALL THE OFFLINE VERSION OF TWS) - anyway as I was saying, once you've installed TWS and unpacked the IBController zip file, all you have to do is make sure IBControllerStart.[ ini | sh ] contains the correct TWS major version number and the correct location for IBController.ini, for example (on Windows):

`set TWS_MAJOR_VRSN=952`
`set IBC_INI=%HOMEPATH%\Documents\IBController\IBController.ini`

That's all! How hard was that! Then it will work, guaranteed! It really is that simple. You don't have to change _anything_ else to get a basic working system. Once you've got that, you can start modifying other options to tailor the system to your heart's desire.

So let's talk about the error message you showed, or rather the class of such error messages (I've never seen this particular one before and I don't have the slightest idea how you managed to make it happen). The line of code in IBController where this occurs says simply this:

`jclient.LoginFrame.main(twsArgs);`

Here IBController is simply trying to enter the TWS code. It has no idea which .jar file contains that code, and indeed this has changed completely recently. The Java runtime takes care of loading the appropriate class from whatever .jar it's in and transferring control to it.

Now I could certainly put a try...catch block around that line, but about the only sensible error message I could give would be 

`Can't find the TWS entry point: jclient.LoginFrame.main. TWS is not properly installed.`

So do you think such an error message would save a lot of your lifetime? If you do, I'm perfectly happy to include it in the next IBController release (in fact I've just done so). But it's really not, in my opinion, a lot more helpful than the low-level verbiage in the message you showed. The trouble is IBController has absolutely no idea whatever what might have gone wrong here, and there is nothing it can check as you suggest.

**Comment by rlktradewright** [comment]
This is addressed in Release 3.2.0-beta.1 issued today.
